### PR TITLE
Add support for DS3231 Extrnal RTC with I2c interface

### PR DESCRIPTION
**Pull request by guyelg**
Tested with nodeMcu 1.0.
sensor number 33

### CODE REVIEW

File: sonoff/user_config.h [hunk 1]
```diff
@@ -322,6 +322,8 @@
       #define MTX_ADDRESS6     0x76              // [DisplayAddress6] I2C address of sixth 8x8 matrix module
       #define MTX_ADDRESS7     0x00              // [DisplayAddress7] I2C address of seventh 8x8 matrix module
       #define MTX_ADDRESS8     0x00              // [DisplayAddress8] I2C address of eigth 8x8 matrix module
+  #define USE_DS3231                          // Enable use DS3231 external RTC , usefall when you don't have avaliable WIFI. see docs in the source file (+1k2 code)
```
Comment (ascillato) [inline sonoff/user_config.h]: This need to be commented. By default it is not being used by sonoff devices.

File: sonoff/user_config.h [hunk 2]
```diff
@@ -322,6 +322,8 @@
...
+//  #define USE_RTC_ADDR                      //you can change the addrsss of the DS3231 RTC, default is 0x68, not mandatory fieled
```
Comment (ascillato) [inline sonoff/user_config.h]: Please, add at least the default address to be left as an example for address syntax. If left alone can be confused with an option instead that a number required.

**Review by ascillato — CHANGES_REQUESTED** [review]

File: sonoff/user_config.h (hunk 2 above)
Comment (andrethomas) [reply sonoff/user_config.h]: Another option is to combine the two

`#define USE_DS3231   0x68`

**Comment by ascillato** [comment]
Hi,

By default this driver needs to be disabled. Please, update that for user_config.h

**Comment by ascillato** [comment]
This solves #2871

**Review by ascillato — APPROVED** [review]

**Review by andrethomas — APPROVED** [review]

**Comment by ascillato** [comment]
@guyelg

Thanks :+1:

**Pull request merged**